src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,240 @@
+
+```
+string[] dishes = { "sushi", "pasta", "cucumber soup" };
+IEnumerable<string> sports = sportsExperienceYears.Where(sport => ((JsonNumber)sport.Value).GetInt32() > 2).Select(sport => sport.Key);
+
+var preferences = new JsonObject()
+{
+    { "colours", new JsonArray { "red", "green", "purple" } },
+    { "numbers", new JsonArray { 4, 123, 88 } },
+    { "prime numbers", new JsonNumber[] { 19, 37 } },
+    { "dishes", new JsonArray(dishes) },
+    { "sports", new JsonArray(sports) },
+    { "strange words", strangeWords.Where(word => ((JsonString)word).Value.Length < 10) },
+};
+```
+
+### Modifying existing instance
+
+The main goal of the new API is to allow users to modify existing instance of `JsonNode` which is not possible with `JsonElement` and `JsonDocument`. 
+
+One may change the existing property to have a different value:
+```
+ var options = new JsonObject { { "use caching", true } };
+ options["use caching"] = (JsonBoolean)false;
+```
+
+Add a value to existing JSON array or property to existing JSON object:
+```
+var bestEmployees = new JsonObject(EmployeesDatabase.GetTenBestEmployees());
+bestEmployees.Add(EmployeesDatabase.GetManager());
+```
+
+Or modify the exisitng property name:
+```
+JsonObject manager = EmployeesDatabase.GetManager();
+JsonObject reportingEmployees = manager.GetJsonObjectProperty("reporting employees");
+reportingEmployees.ModifyPropertyName("software developers", "software engineers");
+```
+
+### Transforming to and from JsonElement
+
+The API allows users to get a writable version of JSON document from readonly one and vice versa:
+
+Transforming JsoneNode to JsonElement:
+```
+JsonNode employeeDataToSend = EmployeesDatabase.GetNextEmployee().Value;
+Mailbox.SendEmployeeData(employeeDataToSend.AsJsonElement());
+```
+
+Transforming JsonElement to JsonNode:
+```
+JsonNode receivedEmployeeData = JsonNode.DeepCopy(Mailbox.RetrieveMutableEmployeeData());
+if (receivedEmployeeData is JsonObject employee)
+{
+    employee["name"] = new JsonString("Bob");
+}
+```
+
+### Parsing to JsonNode
+
+If developer knows he/she wil be modifying and instance, API provides the possibility to parse string rigth to `JsonNode`, without `JsonDocument` being an intermediary.
+
+```
+string jsonString = @"
+{
+    ""employee1"" : 
+    {
+        ""name"" : ""Ann"",
+        ""surname"" : ""Predictable"",
+        ""age"" : 30,                
+    },
+    ""employee2"" : 
+    {
+        ""name"" : ""Zoe"",
+        ""surname"" : ""Coder"",
+        ""age"" : 24,                
+    }
+}";
+
+JsonObject employees = JsonNode.Parse(jsonString) as JsonObject;
+employees.Add(EmployeesDatabase.GetNextEmployee());
+Mailbox.SendAllEmployeesData(employees.AsJsonElement());
+```
+
+## Desingn choices
+
+* No advanced methods for looking up properties like `GetAllValuesByPropertyName` or `GetAllPrimaryTypedValues`, because they would be too specialized.
+* `null` reference to node instead of `JsonNull` class.
```

Comment:
It just seemed redundant. Here's a comment addressing it: https://github.com/kasiabulat/corefx/pull/1#discussion_r304018769. 
And it is not possible to write `new JsonObject{ {"null_property", null} }`, because it's ambiguous, so it has to be `new JsonObject{ {"null_property", (JsonNode) null} }`